CanvasColorSpaceProposal.md

```diff
@@ -63,56 +104,72 @@ partial interface CanvasRenderingContext2D {
+* toDataURL/toBlob produce resources tagged as being in the sRGB color space, if the encoding format supports colorspace tagging or embedded color profiles.
+    * Exception: User agents that implement the legacy behavior must not encode any color space metadata.
+
+##### The "rec-2020" color space
+* As per the [CSS rec-2020 color space](https://www.w3.org/TR/css-color-4/#valdef-color-rec-2020).
+* This color space has different primaries and a different transfer function than "srgb".
+* Support is optional, and should not be present on srgb-only UAs.
+
+
+#### The colorEncoding context creation attribute
+The colorEncoding attributes specifies the encoding to be used for storing pixel channel color values.
+* Support for "unorm8" is mandatory. All other encodings are optional.
+* When an unsupported encoding is requested, the encoding shall fall back to "unorm8".
+* The alpha channel is always interpreted as if clamped to [0,1].
+* Float RGB channel values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a luminance-linear pixel value of (2,2,2) should be twice as bright as (1,1,1).
+* Operations on encoded values always operate on decoded values, not the encoded bits.
+    * I.e. in "unorm8-srgb" encoding, 0xff (1.0) minus 0xbc (0.5) equals 0xbc (0.5).
+
+
+#### The colorLinearity context creation attribute
+
+##### "luminance" linearity
+Values are linear along their transfer functions.
+0.5 is half as (perceptually) luminant/bright as 1.0, but about 20% as (physically) radiant.
+For blending and interpolation, 0.5 is half way between 0.0 and 1.0.
+
+##### "radiance" linearity
+Values are linear with no transfer function.
+0.5 is half as (physically) radiant as 1.0, but about 75% as (perceptually) luminant/bright.
+For blending and interpolation, 0.5 is half way between 0.0 and 1.0.
+
 
 #### Selecting the best color space match for the user agent's display device
 <pre>
-var colorSpace = window.matchMedia("(color-gamut: rec2020)").matches ? "rec2020" : 
+var colorSpace = window.matchMedia("(color-gamut: rec2020)").matches ? "rec2020" :
     (window.matchMedia("(color-gamut: p3)").matches ? "p3" : "srgb");
 </pre>
 
-#### Selecting the best pixelFormat for the user agent's display device
+#### Selecting the best encoding for the user agent's display device
```

Comment:
Even after reading this far I am still having trouble with "encoding" is this context. Too many years of thinking about sRGB encoding. Now that is truly an encoding of the data. Graphics APIs don't refer to encoding. OpenGL Vulkan, DirectX and Metal all have "formats" not encodings. I think you should revert to pixelFormat here.

Reply:
In order to make forward progress on these APIs which are still in the draft stage, could we agree to make the change to encoding / colorEncoding so there's symmetry with colorSpace?